### PR TITLE
CardboardList: Add styling for selected list items

### PR DESCRIPTION
**Pull request by msnyder-msft**
### Summary of changes 🔍 
Adding some styling to mark a list item as selected without needing the full checkbox UI.

image

### Testing 🧪
- [x] Lists --> WithSelectedItem
- [x] Lists --> Items --> WithSelected
 
### Checklist ✔️
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: src/Components/CardboardList/CardboardListItem.tsx [hunk 1]
```diff
@@ -42,7 +43,6 @@ export const CardboardListItem = <T extends unknown>(
     const showWarningIndicator = isValid === false;
 
     // end icon
-    const isEndIconCustomRender = typeof iconEnd !== 'object';
```
Comment (pasanchMSFT) [inline src/Components/CardboardList/CardboardListItem.tsx:45]: Why was this removed?

Comment (msnyder-msft) [reply src/Components/CardboardList/CardboardListItem.tsx:45]: cause it turns out I only used it in that useEffect and it wasn't actually used in there anymore. Woops.

**Review by pasanchMSFT — APPROVED** [review]

**Review by iremgokce — APPROVED** [review]

**Review by montgomp — APPROVED** [review]
